core/kazoo_globals/test/kz_globals_pqc.erl

```diff
@@ -0,0 +1,292 @@
+        'true' -> Registries;
+        'false' ->
+            maps:fold(fun(Node, Registry, Acc) when Remote =:= Node ->
+                              Acc#{Node => [?REG(Name, Pid, 'local') | Registry]};
+                         (Node, Registry, Acc) ->
+                              Acc#{Node => [?REG(Name, Pid, 'remote') | Registry]}
+                      end
+                      ,maps:new()
+                      ,Registries
+                     )
+    end;
+next_state(Registries, _V
+           ,{'call', ?MODULE, 'remote_unregister', [Remote, Name]}
+          ) ->
+    #{Remote := Registry} = Registries,
+    case lists:keysearch(Name, 1, Registry) of
+        'false' -> Registries;
+        {'value', ?REG(Name, _P, 'local')} ->
+            unregister_name(Name, Registries);
+        {'value', _} -> Registries
+    end;
+
+%% Local Node State Changes
+next_state(Registries, _V
+          ,{'call', 'kz_globals', 'whereis_name', [_Name]}
+          ) ->
+    Registries;
+next_state(Registries, _V
+          ,{'call', 'kz_globals', 'register_name', [Name, Pid]}
+          ) ->
+    case name_exists(Name, Registries) of
+        'true' -> Registries;
+        'false' ->
+            maps:fold(fun(Node, Registry, Acc) when ?LOCAL_NODE =:= Node ->
+                              Acc#{Node => [?REG(Name, Pid, 'local') | Registry]};
+                         (Node, Registry, Acc) ->
+                              Acc#{Node => [?REG(Name, Pid, 'remote') | Registry]}
+                      end
+                     ,maps:new()
+                     ,Registries
+                     )
+    end;
+next_state(#{?LOCAL_NODE := LocalRegistry}=Registries, _V
+          ,{'call', 'kz_globals', 'unregister_name', [Name]}
+          ) ->
+    case lists:keysearch(Name, 1, LocalRegistry) of
+        'false' -> Registries;
+        {'value', ?REG(Name, _Pid, 'local')} ->
+            unregister_name(Name, Registries);
+        {'value', ?REG(Name, _Pid, _State)} -> Registries
+    end;
+next_state(Registries, _V
+          ,{'call', 'kz_globals', 'registered', []}
+          ) ->
+    Registries.
+
+precondition(Registries
+             ,{'call', ?MODULE, 'remote_register', [Node, Name, _Pid]}
+            ) ->
+    #{Node := Registry} = Registries,
+    case lists:keysearch(Name, 1, Registry) of
+        {'value', ?REG(Name, _P, _State)} -> 'false';
+        'false' -> 'true'
+    end;
+precondition(Registries
+             ,{'call', ?MODULE, 'remote_unregister', [Node, Name]}
+            ) ->
+    #{Node := Registry} = Registries,
+    case lists:keysearch(Name, 1, Registry) of
+        {'value', ?REG(Name, _P, 'local')} -> 'true';
+        _ -> 'false'
+    end;
+precondition(#{?LOCAL_NODE := Registry}
+             ,{'call', 'kz_globals', 'unregister_name', [Name]}
+            ) ->
+    case lists:keysearch(Name, 1, Registry) of
+        {'value', ?REG(Name, _Pid, 'local')} -> 'true';
+        _ -> 'false'
+    end;
```

Comment:
The nodes won't unregsiter_name if they haven't registered the name already. These conditions check the node's registry to see if calling unregister_name makes sense.